packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -125,11 +164,110 @@ class TlsValidationFileTrust extends TlsValidationTrust {
           certificateChain: this.certificateChain,
         },
       },
+      virtualNodeListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
       virtualGatewayClientTlsValidationTrust: {
         file: {
           certificateChain: this.certificateChain,
         },
       },
+      virtualGatewayListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
     };
   }
 }
+
+class TlsValidationSdsTrust extends TlsValidationTrust {
+  /**
+   * The name of the secret for Envoy to fetch from a specific endpoint through the Secrets Discovery Protocol.
+   */
+  readonly secretName: string;
+
+  constructor (props: TlsValidationSdsTrustOptions) {
+    super();
+    this.secretName = props.secretName;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    return {
+      virtualNodeClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualNodeListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayClientTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
+      virtualGatewayListenerTlsValidationTrust: {
+        sds: {
+          secretName: this.secretName,
+        },
+      },
```

Comment:
You can reduce the repetition (mentioned this on the last PR I think). Example: https://github.com/alexbrjo/aws-cdk/commit/aea876e1d62fb34437a774c9861bdcd9c6b91527

Reply:
Reduced the repetition to simplify the code.